app/src/main/java/com/github/mik629/android/fundamentals/data/db/models/MovieGenreCrossRef.kt

```diff
@@ -0,0 +1,25 @@
+package com.github.mik629.android.fundamentals.data.db.models
+
+import androidx.room.Entity
+import androidx.room.ForeignKey
+import androidx.room.PrimaryKey
+import com.github.mik629.android.fundamentals.data.db.contracts.DbContract.Genres.COLUMN_NAME_GENRE_ID
+import com.github.mik629.android.fundamentals.data.db.contracts.DbContract.Movies.COLUMN_NAME_MOVIE_ID
+
+@Entity
+data class MovieGenreCrossRef(
+    @PrimaryKey(autoGenerate = true)
+    val id: Int = 0,
```

Comment:
Nope, room does, I believe

Reply:
https://developer.android.com/training/data-storage/room/relationships#many-to-many

Reply:
Been there. When I defined it like this then I got a duplicate entity detected. Seems like that happened because of inserts didn't have conflict resolution by that time. Now it works, thanks